src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -115,6 +117,24 @@ std::vector<TRShape> shape_infer(const Squeeze* op,
     }
     return output_shapes;
 }
+}  // namespace util
+
+namespace v0 {
+template <class T, class TRShape = result_shape_t<T>>
+std::vector<TRShape> shape_infer(const Squeeze* op,
+                                 const std::vector<T>& input_shapes,
+                                 const ITensorAccessor& ta = make_tensor_accessor()) {
+    return util::shape_infer(op, input_shapes, ta);
+}
 }  // namespace v0
+
+namespace v15 {
+template <class T, class TRShape = result_shape_t<T>>
+std::vector<TRShape> shape_infer(const Squeeze* op,
+                                 const std::vector<T>& input_shapes,
+                                 const ITensorAccessor& ta = make_tensor_accessor()) {
+    return util::shape_infer(op, input_shapes, ta);
```

Comment:
Shape inference for v15 is different than v0. There is additional attribute which should be handled here only.

Reply:
Agree, please use the `shape_infer` implementation from the POC PR, and apply it for v15::Squeeze shape_infer only.
If possible, update it to skip this private Squeeze `deduced_output_shape` member (discussed offline).
 
https://github.com/openvinotoolkit/openvino/pull/25774/files#diff-23e1a31f71a4d9f2d3a6c63bdb363f5b022b4548fe7419efef1b889bb0b39b4fR16

Reply:
Done